lox_services/proxy/brightdata.py

```diff
@@ -69,122 +76,134 @@ def _excecute_request_with_retry(request_method: Callable, options: dict, retrie
-            - `countries`: A list of countries where the proxies should be.
-            - `number_of_threads`: The number of threads to use.
-            - `max_use_per_proxy`: The maximum number of use per ip for this call.
-            - `log`: If True, the function will print the status of the requests every 10%.
-
-            ## Returns
-            - A list of requests.Response objects
-
-            ## Examples
-            ```python
-            proxy_manager = BrightDataProxyManager(brightdata_username, brightdata_password, brightdata_api_token)
-            test_ips = proxy_manager.request_with_multithreading_proxies(
-                request_method=requests.get,
-                request_options=[{"url": "https://api.ipify.org?format=json"} for _ in range(20)],
-                countries=["FR", "NL"],
-                number_of_threads=20
-            )
-            ```
+        ## Arguments
+        - `request_method`: Must be a requests method like requests.get ot requests.post or requests.put, etc...
+        - `request_options`: The list of options to used by the request_method.
+        - `countries`: A list of countries where the proxies should be.
+        - `number_of_threads`: The number of threads to use.
+        - `max_use_per_proxy`: The maximum number of use per ip for this call.
+        - `log`: If True, the function will print the status of the requests every 10%.
+
+        ## Returns
+        - A list of requests.Response objects
+
+        ## Examples
+        ```python
+        proxy_manager = BrightDataProxyManager(brightdata_username, brightdata_password, brightdata_api_token)
+        test_ips = proxy_manager.request_with_multithreading_proxies(
+            request_method=requests.get,
+            request_options=[{"url": "https://api.ipify.org?format=json"} for _ in range(20)],
+            countries=["FR", "NL"],
+            number_of_threads=20
+        )
+        ```
         """
         number_of_threads = min(50, number_of_threads)
-        proxies = self._get_available_ips_per_countries(countries) 
-        options_with_proxies = self._populate_proxies_into_request_options(request_options, proxies, max_use_per_proxy)
+        proxies = self._get_available_ips_per_countries(countries)
+        options_with_proxies = self._populate_proxies_into_request_options(
+            request_options, proxies, max_use_per_proxy
+        )
         # options_with_proxies = request_options
-        options_len = len(request_options)
-        
+
         results = []
+        options_len = len(request_options)
+        progress_bar = tqdm(total=options_len)
+
         with ThreadPoolExecutor(max_workers=number_of_threads) as executor:
-            for result in executor.map(self._excecute_request_with_retry, repeat(request_method), options_with_proxies):
-                if log and len(results) % max(1, int(options_len / 10)) == 0:
-                    print(f"{len(results)} / {options_len}")
-                
```

Comment:
I agree that this is nice to see the evolution of the scrapping
However, or UPS, we call this function per batch of 400 tracking numbers and we scrape more than 1M per month, this will increase by a lot the time the script will take
Can we have this bar only if display = True?

Reply:
Done.